src/Umbraco.Web/Unversion/UnversionConfig.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Web;
+using System.Xml;
+using Umbraco.Core.Logging;
+
+namespace Umbraco.Web.Unversion
+{
+    public class UnversionConfig : IUnversionConfig
+    {
+        public const string AllDocumentTypesKey = "$_ALL";
+        public IDictionary<string, List<UnversionConfigEntry>> ConfigEntries { get; set; }
+
+        private ILogger _logger;
+
+        public UnversionConfig(ILogger logger)
+        {
+            _logger = logger;
+
+            ConfigEntries = new Dictionary<string, List<UnversionConfigEntry>>();
+
+            try
+            {
+                var appPath = HttpRuntime.AppDomainAppPath;
+                var configFilePath = Path.Combine(appPath, @"config\unversion.config");
+                LoadXmlConfig(string.Concat(configFilePath));
+            }
+            catch (Exception e)
+            {
+                _logger.Error<UnversionConfig>(e, "Error when parsing unversion.config.");
+            }
+
+        }
+
+        private void LoadXmlConfig(string configPath)
+        {
+            if (!File.Exists(configPath))
+            {
+                _logger.Warn<UnversionConfig>("Couldn't find config file {configPath}", configPath);
+                return;
+            }
+
+            var xmlConfig = new XmlDocument();
+            xmlConfig.Load(configPath);
+
+            foreach (XmlNode xmlConfigEntry in xmlConfig.SelectNodes("/unversionConfig/add"))
+                if (xmlConfigEntry.NodeType == XmlNodeType.Element)
+                {
+                    var configEntry = new UnversionConfigEntry
+                    {
+                        // If no doctype alias set on XML entry then its assume it's for all doctypes
+                        DocTypeAlias = xmlConfigEntry.Attributes["docTypeAlias"] != null
+                            ? xmlConfigEntry.Attributes["docTypeAlias"].Value
+                            : AllDocumentTypesKey
+                    };
+
+                    if (xmlConfigEntry.Attributes["rootXpath"] != null)
+                        configEntry.RootXPath = xmlConfigEntry.Attributes["rootXpath"].Value;
+
+                    if (xmlConfigEntry.Attributes["maxDays"] != null)
+                        configEntry.MaxDays = Convert.ToInt32(xmlConfigEntry.Attributes["maxDays"].Value);
+
+                    if (xmlConfigEntry.Attributes["maxCount"] != null)
+                        configEntry.MaxCount = Convert.ToInt32(xmlConfigEntry.Attributes["maxCount"].Value);
```

Comment:
For example, this will just throw an ugly exception if it's not formatted correctly, it's not being proactive in checking for correct data formats. Again, this would be solved by real .net configuration classes like we have for everything else.